spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,990 @@
+        policy_group = @vsd_client.get_policy_group(POLICY_GROUP_ID)
+      end
+
+      assert_object_not_empty(policy_group)
+      assert_policy_group(policy_group)
+    end
+
+    it "should return valid non empty response for policy_groups for domain" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_for_domain', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups_for_domain(DOMAIN_ID)
+      end
+
+      assert_object_not_empty(policy_groups)
+      assert_policy_groups(policy_groups)
+    end
+
+    it "should return empty response for policy_groups" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_empty', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups
+      end
+      expect(policy_groups).to be_nil
+    end
+
+    it "should return empty response for policy_group by id" do
+      policy_group = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_group_empty', :record => :new_episodes) do
+        policy_group = @vsd_client.get_policy_group(POLICY_GROUP_ID)
+      end
+      expect(policy_group).to be_nil
+    end
+
+    it "should return empty response for policy_groups for domain" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_for_domain_empty', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups_for_domain(DOMAIN_ID)
+      end
+      expect(policy_groups).to be_nil
+    end
+  end
+
+  it "should fail on wrong password" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_pass', :record => :new_episodes) do
+      expect do
+        described_class.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, 'wrong_password')
+      end.to raise_error(MiqException::MiqInvalidCredentialsError)
+    end
+  end
+
+  it "should fail on wrong username" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_user', :record => :new_episodes) do
+      expect do
+        described_class.new("https://#{@hostname}:8443/nuage/api/v5_0", 'wrong_user', @password)
+      end.to raise_error(MiqException::MiqInvalidCredentialsError)
+    end
+  end
+
+  it "should fail on wrong hostname" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_hostname', :record => :new_episodes) do
+      expect do
+        described_class.new("https://wronghost:8443/nuage/api/v5_0", @userid, @password)
+      end.to raise_error(SocketError)
+    end
+  end
+
+  def assert_object_not_empty(object)
+    expect(object.length).to be > 0
+    expect(object.length).to be_truthy
+  end
+
+  def assert_enterprises(enterprises)
+    # test count
```

Comment:
From what I've seen, we're generally supposed to comment code only when really needed. I'd remove two comments here since it's obvious what we're doing.

Reply:
Also, I'm not sure that you need to check the count here again since you check it in each `it` explicitly by the means of `assert_object_not_empty`.

Reply:
that counts number of objects in the list, the check in `assert_object_not_empty` checks if the object is empty (it shouldn't be)